### ISSUE TITLE
Chromecast with Google TV

### ISSUE THREAD

**Issue by philled** [MythTV/packaging#99]
I'm not able try try compiling my own APK. I've been trying to get the publicly available  pre-compiled APKs to work on my Chromecast with Google TV. Every APK I could find fails to authenticate with the backend database. The Mythtv mailing list thinks this may be because the APKs were compiled using an out of date version of the MySQL client library.

Does anyone have an APK that they know works on Chromecast with Google TV that they would be willing to share?

**Comment by philled** [comment]
>
> The apks on orangedox.com work with android tv so they could ne be
> compiled with a wrong client library.
> Please explain in detail what happens when you run mythfrontend.
>
Hi, when I open the mythfrontend app it finds the backend server and
populates the IP address, database name (mythconverg) and user (mythtv).
I'm 100% certain I've entered the correct password. But it won't connect to
the database. I have a linux frontend on the same subnet which connects
fine. I've tested that the MySQL connection requests are going through
using tshark to monitor the traffic coming into the mythbackend machine and
it shows MySQL returning error 1158:

 147 88.911855731 192.168.0.120 → 192.168.0.125 MySQL 161 Server Greeting
proto=10 version=8.0.26-0ubuntu0.20.04.2
  148 88.912727650 192.168.0.125 → 192.168.0.120 TCP 66 39502 → 3306 [ACK]
Seq=1 Ack=96 Win=87808 Len=0 TSval=131226266 TSecr=1186449381
  149 88.915360803 192.168.0.125 → 192.168.0.120 MySQL 232 Login Request
user=mythtv db=mythconverg
  150 88.915367067 192.168.0.120 → 192.168.0.125 TCP 66 3306 → 39502 [ACK]
Seq=96 Ack=167 Win=65152 Len=0 TSval=1186449385 TSecr=131226267
  151 88.915469028 192.168.0.120 → 192.168.0.125 MySQL 114 Auth Switch
Request
  152 88.917330547 192.168.0.125 → 192.168.0.120 TCP 66 39502 → 3306 [FIN,
ACK] Seq=167 Ack=144 Win=87808 Len=0 TSval=131226267 TSecr=1186449385
  153 88.917457270 192.168.0.120 → 192.168.0.125 MySQL 121 Response Error
1158
  154 88.917470722 192.168.0.120 → 192.168.0.125 TCP 66 3306 → 39502 [FIN,
ACK] Seq=199 Ack=168 Win=65152 Len=0 TSval=1186449387 TSecr=131226267
  155 88.918298535 192.168.0.125 → 192.168.0.120 TCP 60 39502 → 3306 [RST]
Seq=168 Win=0 Len=0
  156 88.918774149 192.168.0.125 → 192.168.0.120 TCP 60 39502 → 3306 [RST]
Seq=168 Win=0 Len=0
  157 89.996525845 34:0a:33:87:ad:00 → Broadcast    ARP 60 Who has
192.168.0.125? Tell 192.168.0.204
  158 89.999819001 192.168.0.125 → 192.168.0.120 TCP 74 39504 → 3306 [SYN]
Seq=0 Win=65535 Len=0 MSS=1460 SACK_PERM=1 TSval=131226538 TSecr=0 WS=256
  159 89.999861548 192.168.0.120 → 192.168.0.125 TCP 74 3306 → 39504 [SYN,
ACK] Seq=0 Ack=1 Win=65160 Len=0 MSS=1460 SACK_PERM=1 TSval=1186450469
TSecr=131226538 WS=128
  160 90.000509424 192.168.0.125 → 192.168.0.120 TCP 66 39504 → 3306 [ACK]
Seq=1 Ack=1 Win=87808 Len=0 TSval=131226538 TSecr=1186450469

Error 1158 means "Got an error reading communication packets". Stephen
Worthington on the mythtv-users mailing list suggests the problem could be
"that the CGTV app is connecting using an out of date authorisation method.
The MySQL server responds to this with an AuthSwitchRequest packet telling
the CGTV to try a different authorisation method. The CGTV does not like
that and responds by disconnecting, which causes the server to log a
\Response Error 1158'. What that suggests is that the mythfrontend loaded
on the CGTV was compiled using an out of date version of the MySQL client
library."

Could it be possible that the APK has been tested against Maria DB
installations but not a MySQL-based backend and therefore the problem may
not have surfaced until now?

**Comment by bennettpeter** [comment]
I just ran a test to check your theory.
I installed version mythfrontend-20210819-arm-v31.0-163-g5da2523154-dirty.apk on a fire stick 4K. I connected successfully to my v31 backend. The backend is running mysql and checking the version gives the following:

mysql --version
mysql  Ver 8.0.26-0ubuntu0.20.04.2 for Linux on x86_64 ((Ubuntu))

The frontend connects to the backend and runs with no problem. I can get into recordings and navigate the menus with no issue.

So it is not an incompatibility with mysql or with recent versions that is causing the problem.
